sickrage/providers/GenericProvider.py

```diff
@@ -518,3 +519,18 @@ def _make_url(self, result):
 
     def _verify_download(self, file_name=None):  # pylint: disable=unused-argument,no-self-use
         return True
+
+    @property
+    def recent_results(self):
+        if sickbeard.provider_recent_results.get(self.get_id()):
+            return sickbeard.provider_recent_results.get(self.get_id())[::-1]
+        else:
+            return []
+
+    @recent_results.setter
+    def recent_results(self, items):
+        if not sickbeard.provider_recent_results.get(self.get_id()):
+            sickbeard.provider_recent_results.update({self.get_id(): []})
+        sickbeard.provider_recent_results[self.get_id()] += items
+        if items:
+            del sickbeard.provider_recent_results[self.get_id()][:len(sickbeard.provider_recent_results[self.get_id()]) - self.max_recent_items]
```

Comment:
This will cut of the the first n items. For ex. thelist = [1,2,3,4,5,6,7], del thelist[:len(thelist) - 5], will translate to: del thelist[:2] will result in [3,4,5,6,7]. Making sure there are always 5 items in the list.